cmd/tools/cloudburst/cloudburst.go

```diff
@@ -0,0 +1,414 @@
+			Name:            b.testID,
+			Description:     fmt.Sprintf("Load Test Group for Test %s", b.testID),
+			APISecurityLock: false,
+		})
+	if err != nil {
+		return errors.Wrapf(err, "failed to create a group for test %s", b.testID)
+	}
+	b.group = group
+	return nil
+}
+
+// cleanupInstallations attempts to delete the created Installations
+// until it succeeds, and blocks until that time.
+func (b *Blaster) cleanupInstallations(installations map[string]*cloud.Installation) {
+	for len(installations) > 0 {
+		for _, install := range installations {
+			fetched, err := b.client.GetInstallation(install.ID, nil)
+			if err != nil {
+				logger.WithError(err).Warnf("failed to look up Installation %s", install.ID)
+				continue
+			}
+			if fetched == nil {
+				logger.Warnf("Installation %s not found; will not retry deletion", install.ID)
+				b.addReports(&errorReport{
+					installation: install,
+					timestamp:    time.Now(),
+					message:      fmt.Sprintf("%s not found", install.ID),
+				})
+				delete(installations, install.ID)
+				continue
+			}
+			switch fetched.State {
+			case cloud.InstallationStateDeleted:
+				logger.Infof("Successfully deleted Installation %s", fetched.ID)
+				delete(installations, fetched.ID)
+			case cloud.InstallationStateDeletionRequested:
+				fallthrough
+			case cloud.InstallationStateDeletionFinalCleanup:
+				fallthrough
+			case cloud.InstallationStateDeletionInProgress:
+			default:
+				b.client.DeleteInstallation(fetched.ID)
+			}
+		}
+	}
+}
+
+// cleanupGroup just deletes the group and returns an error if it
+// fails. It does not retry or block.
+func (b *Blaster) cleanupGroup() error {
+	err := b.client.DeleteGroup(b.group.ID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete group %s", b.group.ID)
+	}
+	return nil
+}
+
+// createInstallations will request creation of "total" installations
+// in parallel, created in serial batches of size batchSize. A
+// batchSize of 1 will deploy all installations in parallel, and a
+// batchSize equal to total will deploy them all serially.
+func (b *Blaster) createInstallations(total, batchSize int) map[string]*cloud.Installation {
+	installationsChannel := make(chan []*cloud.Installation)
+	for i := 0; i < total; i += batchSize {
+		go func(out chan []*cloud.Installation, batchNumber int) {
+			batch := b.serialBatchInstall(batchSize)
+			out <- batch
+		}(installationsChannel, i)
+	}
+
+	allInstallations := map[string]*cloud.Installation{}
+	for i := 0; i < total/batchSize; i++ {
+		batch := <-installationsChannel
+		for _, installation := range batch {
+			allInstallations[installation.ID] = installation
+		}
+	}
+
+	return allInstallations
+}
+
+// createInstallation requests creation of a single installation and
+// returns an error if the API does. It does not block or retry.
+func (b *Blaster) createInstallation() (*cloud.Installation, error) {
+	installationDTO, err := b.client.CreateInstallation(
+		&cloud.CreateInstallationRequest{
```

Comment:
Being able to specify `version`, `filestore`, and `database` with flags in the future would be nice.

Reply:
https://mattermost.atlassian.net/browse/MM-30529 Created this story to track this work in the future